packages/test/test-end-to-end-tests/src/test/gc/gcTombstoneDataStores.spec.ts

```diff
@@ -787,31 +802,46 @@ describeNoCompat("GC data store tombstone tests", (getTestObjectProvider) => {
 					url: unreferencedId,
 					headers: { [AllowTombstoneRequestHeaderKey]: true },
 				};
-				const tombstonedObject = await requestFluidObject<ITestDataObject>(
-					tombstoneContainer,
-					requestAllowingTombstone,
-				);
+				let tombstonedObject: ITestDataObject;
+				{
+					const response = await containerRuntime.resolveHandle(requestAllowingTombstone);
+					tombstonedObject = response.value as ITestDataObject;
+				}
 				const defaultDataObject =
 					(await tombstoneContainer.getEntryPoint()) as ITestDataObject;
 				defaultDataObject._root.set("store", tombstonedObject.handle);
 
 				// The datastore should be un-tombstoned now
 				const { summaryVersion: revivalVersion } = await summarize(summarizer);
 				const revivalContainer = await loadContainer(revivalVersion);
-				const revivedObject = await requestFluidObject<ITestDataObject>(
-					revivalContainer,
-					unreferencedId,
-				);
+				let revivedObject: ITestDataObject;
+				{
+					const revivalEntryPoint =
+						(await revivalContainer.getEntryPoint()) as ITestDataObject;
+					const revivalContainerRuntime = revivalEntryPoint._context
+						.containerRuntime as ContainerRuntime;
+					const response = await revivalContainerRuntime.resolveHandle({
+						url: unreferencedId,
+					});
+					revivedObject = response.value as ITestDataObject;
+				}
 				revivedObject._root.set("can send", "op");
 				// This signal call closes the tombstoneContainer.
 				// The op above doesn't because the signal reaches the tombstone container faster
 				revivedObject._runtime.submitSignal("can submit", "signal");
 
 				const sendingContainer = await loadContainer(revivalVersion);
-				const sendDataObject = await requestFluidObject<ITestDataObject>(
-					sendingContainer,
-					unreferencedId,
-				);
+				let sendDataObject: ITestDataObject;
+				{
+					const sendingEntryPoint =
+						(await sendingContainer.getEntryPoint()) as ITestDataObject;
+					const sendingContainerRuntime = sendingEntryPoint._context
+						.containerRuntime as ContainerRuntime;
+					const response = await sendingContainerRuntime.resolveHandle({
```

Comment:
nit: Use `resolveHandleHelper` here? Maybe you can udpate `resolveHandleHelper` to accept the container instead of container runtime and move the logic there.

Reply:
I see, you are also using it for the summarizer. never mind.

Reply:
I created that helper strictly to retain the testing around `assert.throws` and `assert.doesNotThrow`. It provides no value for this case